app/components/ConfirmTransaction/index.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import HdKey from 'ethereumjs-wallet/hdkey'
+import Bip39 from 'bip39'
+import CryptoJs from 'crypto-js'
+
+import 'components/App/styles.css'
+import { assignTransaction } from '../../actions/transactions'
+
+class ConfirmTransaction extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      password: '',
+      errorMessage: '',
+      transaction: {},
+    }
+  }
+
+  componentWillMount = () => {
+    chrome.windows.getCurrent((window) => {
```

Comment:
Have you tried to do not place those lines here and move them to the constructor? To me, when the component is going to be mounted, the transaction info is available, so maybe there is no need to use the ```componentWillMount``` hook.

Also if in the constructor and in the ```componentWillMount```hook the state should be created using direct assignment -> ```this.state = ... ```

See:
https://www.andreasreiterer.at/web-development/reactjs-lifecycle-methods
https://engineering.musefind.com/react-lifecycle-methods-how-and-when-to-use-them-2111a1b692b1